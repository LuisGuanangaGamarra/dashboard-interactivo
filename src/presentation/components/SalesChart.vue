<template>
  <div class="bg-white dark:bg-gray-800 p-6 rounded-2xl shadow">
    <h3 class="text-lg font-bold mb-4">Ventas por mes</h3>
    <div class="h-72">
      <Line :data="chartData" :options="chartOptions" />
    </div>
  </div>
</template>


<script setup lang="ts">
import {
  Chart as ChartJS,
  Title,
  Tooltip,
  Legend,
  LineElement,
  CategoryScale,
  LinearScale,
  PointElement,
} from 'chart.js'
import { Line } from 'vue-chartjs'


ChartJS.register(Title, Tooltip, Legend, LineElement, CategoryScale, LinearScale, PointElement)


const chartData = {
  labels: ['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun'],
  datasets: [
    {
      label: 'Ventas',
      data: [1500, 3200, 2800, 3500, 4000, 4600],
      borderColor: '#3b82f6',
      backgroundColor: 'rgba(59, 130, 246, 0.2)',
      fill: true,
    },
  ],
}


const chartOptions = {
  responsive: true,
  maintainAspectRatio: false,
}
</script>