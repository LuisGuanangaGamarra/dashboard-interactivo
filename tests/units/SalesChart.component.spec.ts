import { mount } from '@vue/test-utils'
import SalesChart from '@/presentation/components/SalesChart.vue'
import { describe, it, expect, vi } from 'vitest'

vi.mock('vue-chartjs', () => ({
    Line: {
        name: 'LineChartMock',
        props: ['data', 'options'],
        template: '<div class="line-chart-mock" />',
    },
}))

describe('SalesChart', () => {
    beforeEach(() => {
        vi.clearAllMocks()
    })

    it('should render the component properly', () => {
        const wrapper = mount(SalesChart)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render the title "Ventas por mes"', () => {
        const wrapper = mount(SalesChart)
        expect(wrapper.text()).toContain('Ventas por mes')
    })

    it('should pass the correct chart data to the Line chart', () => {
        const wrapper = mount(SalesChart)
        const chart = wrapper.findComponent({ name: 'LineChartMock' })

        expect(chart.props('data')).toEqual({
            labels: ['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun'],
            datasets: [
                {
                    label: 'Ventas',
                    data: [1500, 3200, 2800, 3500, 4000, 4600],
                    borderColor: '#3b82f6',
                    backgroundColor: 'rgba(59, 130, 246, 0.2)',
                    fill: true,
                },
            ],
        })
    })

    it('should pass the correct chart options to the Line chart', () => {
        const wrapper = mount(SalesChart)
        const chart = wrapper.findComponent({ name: 'LineChartMock' })

        expect(chart.props('options')).toEqual({
            responsive: true,
            maintainAspectRatio: false,
        })
    })

})