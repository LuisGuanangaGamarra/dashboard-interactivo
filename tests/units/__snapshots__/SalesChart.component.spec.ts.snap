// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SalesChart > should render the component properly 1`] = `
"<div class="sales-chart bg-white dark:bg-gray-800 p-6 rounded-2xl shadow">
  <h3 class="text-lg font-bold mb-4">Ventas por mes</h3>
  <div class="h-72">
    <div class="line-chart-mock"></div>
  </div>
</div>"
`;
